src/LayerPlan.cpp

```diff
@@ -749,9 +754,30 @@ void LayerPlan::addWall(ConstPolygonRef wall, int start_idx, const GCodePathConf
             computeDistanceToBridgeStart((start_idx + point_idx - 1) % wall.size());
         }
 
-        addWallLine(p0, p1, non_bridge_config, bridge_config, flow, non_bridge_line_volume, speed_factor, distance_to_bridge_start);
+        // HACK ALERT
+        // the overlap compensation is not perfect, it can produce short non-flow reduced line segments within a sequence of flow reduced
+        // line segments and so to try and avoid printing the spurious fat line segments we require that their lengths are above a threshold
+
+        const coord_t max_spurious_fat_segment_length = 50; // microns
+
+        if (flow >= wall_min_flow && (first_line || !travel_required || vSize(p0 - p1) > max_spurious_fat_segment_length))
```

Comment:
`addWallLine` is not called if the line is smaller than 50 microns now (unless it's the first line).

Reply:
That's true - I suppose you could get a first line printed that is shorter than 50 microns that you would otherwise not print but I don't think that is very likely and even if you did print it, so what?

Reply:
This one is still a bit of a problem because its possible to try and print a circle in which all the lines are less than 50 microns. I mean, unlikely because people often post files with stupidly low resolutions online, but it could happen. Then we'd end up skipping all the entire shape.

Reply:
OK, I'll take the fat line avoidance stuff out for you (and keep it for me).